src/main/java/com/ibm/cloud/cloudant/kafka/transforms/ArrayFlatten.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.kafka.transforms;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.data.ConnectSchema;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.Requirements;
+
+public class ArrayFlatten implements Transformation<SourceRecord> {
+
+    static final String DEFAULT_DELIMITER = ".";
+    private String delimiter = DEFAULT_DELIMITER;
+
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public SourceRecord apply(SourceRecord record) {
+        Object value = record.value();
+        Requirements.requireMapOrNull(value, String.format("%s array flattening SMT", this.getClass().getName()));
+        if (value != null) {
+            return record.newRecord(record.topic(), record.kafkaPartition(), record.keySchema(), record.key(), null, flattenArraysInMap((Map<String, Object>) value), record.timestamp());
+        } else {
+            return record;
```

Comment:
Yes, we need to retain this `null` value record return which is the special case of a tombstone, but we should better handle the other "branch" of `requireMapOrNull`. I need to check on the exact behaviour of:
* `requireMapOrNull`, it may be that we want to catch an exception from there as a place to log/drop the record
* What exactly happens to exceptions from Transformation apply, do they drop the record or stop the whole connector? If it's the former we probably don't need a change.

Reply:
From my previous transform testing, the connector stops and is "stuck" in a crashed state if there's an exception.  Tom shared details with me about how to get it to a "clean" state e.g. terminate app, clear logs, possibly re-create topic